src/Dapr.Actors/Communication/ActorInvokeExceptionData.cs

```diff
@@ -0,0 +1,52 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+namespace Dapr.Actors
+{
+    using System;
+    using System.IO;
+    using System.Runtime.Serialization;
+    using System.Xml;
+    using Dapr.Actors;
+    using Microsoft.Extensions.Logging;
+
+    [DataContract(Name = "ActorInvokeExceptionData", Namespace = Constants.Namespace)]
+    internal class ActorInvokeExceptionData
+    {
+        private static readonly DataContractSerializer ActorInvokeExceptionDataSerializer = new DataContractSerializer(typeof(ActorInvokeExceptionData));
+
+        public ActorInvokeExceptionData(string type, string message)
+        {
+            this.Type = type;
+            this.Message = message;
+        }
+
+        [DataMember]
+        public string Type { get; private set; }
+
+        [DataMember]
+        public string Message { get; private set; }
+
+        internal static ActorInvokeExceptionData Deserialize(Stream buffer)
+        {
+            if ((buffer == null) || (buffer.Length == 0))
+            {
+                return null;
+            }
+
+            using var reader = XmlDictionaryReader.CreateBinaryReader(buffer, XmlDictionaryReaderQuotas.Max);
+            return (ActorInvokeExceptionData)ActorInvokeExceptionDataSerializer.ReadObject(reader);
```

Comment:
As discussed offline - we'll change this to a serializer we're using elsewhere before 1.0 ships.